lang/mrustc/Portfile

```diff
@@ -135,12 +138,28 @@ if {${subport} eq ${name}} {
     build.env-append \
                     PARLEVEL=${build.jobs}
 
+    # macOS before 10.12 needs F_DUPFD_CLOEXEC and clockid_t
+    if {${os.platform} eq "darwin" && ${os.major} < 15} {
+        depends_lib-append \
+                path:lib/libMacportsLegacySupport.dylib:legacy-support
+
+        build.env-replace \
+                    "CXXFLAGS_EXTRA=${configure.cxxflags} ${configure.universal_cxxflags}" \
+                    "CXXFLAGS_EXTRA=${configure.cxxflags} ${configure.universal_cxxflags} -isystem${prefix}/include/LegacySupport"
+
+        build.env-replace \
+                    "LINKFLAGS_EXTRA=${configure.ldflags} ${configure.universal_ldflags}" \
+                    "LINKFLAGS_EXTRA=${configure.ldflags} ${configure.universal_ldflags} -lMacportsLegacySupport"
```

Comment:
I think the preferred method is `[legacysupport::get_library_link_flags]`

Reply:
Yeah, and it poisons LLVM build inside mrustc-rust subport which lead to complicated to track issues.

cleaning everything after PG is much more complicated that this hacks :)